spot-contracts/contracts/PerpetualTranche.sol

```diff
@@ -887,17 +897,43 @@ contract PerpetualTranche is
     }
 
     /// @dev Enforces the total supply and per tranche mint cap. To be invoked AFTER the mint operation.
-    function _enforceMintCaps(ITranche trancheIn) private view {
-        // checks if supply minted using the given tranche is within the cap
-        if (mintedSupplyPerTranche[trancheIn] > maxMintAmtPerTranche) {
-            revert ExceededMaxMintPerTranche();
-        }
-
-        // checks if new total supply is within the max supply cap
+    function _enforceMintCaps(ITranche depositTranche) private view {
+        // Checks if new total supply is within the max supply cap
         uint256 newSupply = totalSupply();
         if (newSupply > maxSupply) {
             revert ExceededMaxSupply();
         }
+
+        // Checks if the value of deposit tranche relative to the other tranches in the reserve
+        // is no higher than the defined limit.
+        //
+        // NOTE: We consider the tranches which are up for rollover,
+        // to be part of the deposit tranche, as given enough time
+        // they will be eventually rolled over into the deposit tranche.
+        // If the reserve has any mature underlying tokens,
+        // we just exempt it from our calculation.
+        IERC20Upgradeable underlying_ = _reserveAt(0);
+        uint256 totalTrancheVal = 0;
+        uint256 depositTrancheValue = 0;
+        uint8 reserveCount = uint8(_reserves.length());
+        for (uint8 i = 1; i < reserveCount; ++i) {
+            ITranche tranche = ITranche(address(_reserveAt(i)));
+            uint256 trancheValue = _computeReserveTrancheValue(
+                tranche,
+                IBondController(tranche.bond()),
+                underlying_,
+                tranche.balanceOf(address(this)),
+                true
```

Comment:
I view `roundUp` behavior to be the exceptional case used in particular situations, like incoming asset values to ensure positive dust.

Since we're just deciding "go/no-go" in this function and dust isn't really an issue, I'd default to the more normal math mode of truncated division.